File: PersianCalendar/src/main/java/com/cepmuvakkit/times/posAlgo/Ecliptic.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.cepmuvakkit.times.posAlgo;

/**
 *
 * @author mehmetrg
 */
public class Ecliptic {

    public double λ; //λ the ecliptic longitude
    public double β; //β the ecliptic latitude
    public double Δ; //distance  in km

    Ecliptic() {
    }

    Ecliptic(double longitude, double latitude) {
        this.λ = longitude;
        this.β = latitude;
    }

    Ecliptic(double longitude, double latitude, double radius) {
        this.λ = longitude;
        this.β = latitude;
        this.Δ = radius;
    }
}

File: PersianCalendar/src/main/kotlin/com/cepmuvakkit/times/posAlgo/Ecliptic.kt
package com.cepmuvakkit.times.posAlgo

import com.byagowi.persiancalendar.ui.astronomy.Zodiac

/**
 * @author mehmetrg
 */
class Ecliptic(
    longitude: Double, // λ: the ecliptic longitude
    latitude: Double, // β: the ecliptic latitude
    radius: Double = 0.0 // Δ: distance in km
) {
    val λ = longitude
    val β = latitude
    val Δ = radius

    val iauZodiac get() = Zodiac.fromIauEcliptic(this)
    val tropicalZodiac get() = Zodiac.fromTropicalEcliptic(this)
}
